credentials/xds/xds.go

```diff
@@ -65,8 +69,21 @@ type credsImpl struct {
 	fallback credentials.TransportCredentials
 }
 
-// handshakeCtxKey is the context key used to store HandshakeInfo values.
-type handshakeCtxKey struct{}
+// handshakeAttrKey is the type used as the key to store HandshakeInfo in
+// the Attributes field of resolver.Address.
+type handshakeAttrKey struct{}
+
+// SetHandshakeInfo returns a copy of attr which is updated with hInfo.
+func SetHandshakeInfo(attr *attributes.Attributes, hInfo *HandshakeInfo) *attributes.Attributes {
+	return attr.WithValues(handshakeAttrKey{}, hInfo)
+}
+
+// GetHandshakeInfo returns a pointer to the HandshakeInfo stored in attr.
+func GetHandshakeInfo(attr *attributes.Attributes) *HandshakeInfo {
```

Comment:
Please make these work on `resolver.Address`es instead of attributes, e.g.:

https://github.com/grpc/grpc-go/blob/759569bb9c835b555736bfb3e2c24f1bc8a6b4ae/internal/hierarchy/hierarchy.go#L35

Reply:
I have changed `SetHandshakeInfo` to work on `resolver.Address` since this will be invoked from the CDS balancer and it has a `resolver.Address` at that point.

`GetHandshakeInfo` is called from the credentials code, and there is no `resolver.Address` at that point.